### PR TITLE
Fix #2048: Added Leaderboard Description

### LINKED ISSUE #2048: Add leaderboard description field
## Current Behaviour
Currently, we don't display leaderboard description under the leaderboard tab.

## Expected Behaviour
A text field which will display the description about the leaderboard.

## Deliverables
- [ ] Add a blank and nullable text field in the Challenge model with the name `leaderboard_description`.
- [ ] Name the migrations as `Add_leaderboard_description_field_in_challenge`
- [ ] Add a card on the frontend above the card for selecting phases to display the leaderboard with the heading `Description`
- [ ] Paste the screenshot of the leaderboard with some sample data in the `Description` card.

### PR DESCRIPTION
**Pull request by vkartik97**
Fixes #2048 
#### Deliverables
- [x]  Added a blank and nullable text field in the Challenge model with the name `leaderboard_description`
- [x]  Named the migrations as `Add_leaderboard_description_field_in_challenge`
- [x]  Added a card on the frontend above the card for selecting phases to display the leaderboard with the heading `Description`
- [x] Added `leaderboard_description` to `challenges/test_view` and `participants/test_view`, assigned with either `None` or Lorem Ipsum Text or didn't assign, to test the constraints. 
- [x] PR to add `leaderboard_description` to EvalAI-Starters - [EvalAI-Starters/pull#4](https://github.com/Cloud-CV/EvalAI-Starters/pull/4)

#### Screenshots

- When `leaderboard_description` is null<br> screenshot from 2019-02-01 00-42-39

- When `leaderboard_description` is not null<br> screenshot from 2019-02-01 00-41-52

### CODE REVIEW

**Comment by vkartik97** [comment]
@deshraj @RishabhJain2018 Please Review

File: apps/challenges/models.py [hunk 1]
```diff
@@ -37,6 +37,7 @@ def __init__(self, *args, **kwargs):
         default=False, verbose_name="Publicly Available", db_index=True)
     enable_forum = models.BooleanField(default=True)
     forum_url = models.URLField(max_length=100, blank=True, null=True)
+    leaderboard_description = models.TextField(null=True, blank=True)
```
Comment (RishabhJain2018) [inline apps/challenges/models.py]: @vkartik97 Can you please mention in comment the default character limit for `TextField`.

**Comment by RishabhJain2018** [comment]
@vkartik97 You also need to delete the migration file and generate again as there a conflict herte https://travis-ci.org/Cloud-CV/EvalAI/builds/505065052#L897

File: apps/challenges/models.py (hunk 1 above)
Comment (vkartik97) [reply apps/challenges/models.py]: @RishabhJain2018 
`models.TextField` have no character limit if not defined. It uses Postgres' Text type which also has no character limit. But
>In any case, the longest possible character string that can be stored is about 1 GB. 
- https://www.postgresql.org/docs/9.6/datatype-character.html
- https://docs.djangoproject.com/en/1.11/ref/models/fields/#textfield

Should I add `max_length` and set it to `2048`?
The Description in screenshot has 602 Characters.

Comment (RishabhJain2018) [reply apps/challenges/models.py]: okay. I think we can leave it as it is.

File: frontend/src/views/web/challenge/leaderboard.html [hunk 2]
```diff
@@ -11,6 +11,16 @@
                         <div></div>
                     </div>
                 </div>
+                <div class="ev-md-container ev-panel-title" ng-if="challenge.page.leaderboard_description">
+                    <div class="row">
+                        <div class="col xs12 s12">
+                            <strong>Description</strong>
+                        </div>
+                        <div class="col s12">
+                            {{challenge.page.leaderboard_description}}
```
Comment (RishabhJain2018) [inline frontend/src/views/web/challenge/leaderboard.html]: Ohh..just noticed it. Can you please use `ng-bind-html` in order to render HTML from the backend.

Comment (vkartik97) [reply frontend/src/views/web/challenge/leaderboard.html]: Thanks for pointing it out : )

**Comment by vkartik97** [comment]
@RishabhJain2018 
Changes in [Using ng-bind-html for Description](https://github.com/Cloud-CV/EvalAI/pull/2056/commits/d9dad79e24010b2042f1f2ad8bd86fa036a11ee4):
- `ng-bind-html` for leaderboard_description
- `md-body-1`(Regular 14px) Font Size for `leaderboard_description` for 15px, as `Description` Title also has size of 15px.

Screenshots:
- `leaderboard_description` = `<b>Bold</b><br>Normal`
Screenshot from 2019-03-13 01-41-57
- Normal Text
Screenshot from 2019-03-13 01-43-52

**Comment by vkartik97** [comment]
Changes in [Using ng-bind-html for Description](https://github.com/Cloud-CV/EvalAI/pull/2056/commits/d9dad79e24010b2042f1f2ad8bd86fa036a11ee4):
- `ng-bind-html` for leaderboard_description
- `md-body-1`(Regular 14px) Font Size for `leaderboard_description` for 15px, as `Description` Title also has size of 15px.

Screenshots:
- `leaderboard_description` = `<b>Bold</b><br>Normal`
Screenshot from 2019-03-13 01-41-57
- Normal Text
Screenshot from 2019-03-13 01-43-52

**Comment by vkartik97** [comment]
@RishabhJain2018 @deshraj Please review!

**Comment by RishabhJain2018** [comment]
@vkartik97 Can you please fix the travis build.

**Comment by vkartik97** [comment]
@RishabhJain2018 @deshraj Please review!

**Pull request merged**